tests/file_system_cache_test.py

```diff
@@ -56,5 +56,48 @@ def test_empty_key_get(self):
      with self.assertRaisesRegex(ValueError , r"key cannot be empty"):
        cache.get("")
 
+  def test_size_of_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      cache.put("foo", b"bar")
+      self.assertEqual(cache._get_cache_directory_size(), 3)
+
+  def test_size_of_empty_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      self.assertEqual(cache._get_cache_directory_size(), 0)
+
+  def test_size_of_existing_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache1 = FileSystemCache(tmpdir)
+      cache1.put("foo", b"bar")
+      del cache1
+      cache2 = FileSystemCache(tmpdir)
+      self.assertEqual(cache2._get_cache_directory_size(), 3)
+
+  def test_cache_is_full(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes = 5)
+      cache.put("foo", b"bar")
+      cache.put("second", b"file")
+      self.assertEqual(cache.get("foo"), None)
+
+  def test_delete_multiple_files(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes = 6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("snd"), None)
+
```

Comment:
Another edge case you should test: inserting a value into the cache that's above the size threshold, even if everything else is deleted. I'm not totally sure what the behavior should be actually! I think it shouldn't insert the value into the cache (to avoid going over the threshold), but it should also _not_ delete everything that's already there.